packages/bp/src/core/dialog/dialog-engine.ts

```diff
@@ -314,12 +314,15 @@ export class DialogEngine {
       }
     }
 
-    if (!timeoutNode || !timeoutFlow) {
-      throw new TimeoutNodeNotFound(`Could not find any timeout node or flow for session "${sessionId}"`)
+    if (timeoutNode && timeoutFlow) {
+      // There is a timeout node and flow, we jump to it
+      this.fillContextForTransition(event, { currentFlow, currentNode, nextFlow: timeoutFlow, nextNode: timeoutNode })
+    } else if (!event.state.context?.hasJumped) {
+      // If there was no jump, we just return the event to have it's state changes persisted
+      return event
     }
 
-    this.fillContextForTransition(event, { currentFlow, currentNode, nextFlow: timeoutFlow, nextNode: timeoutNode })
-
+    // Process the event with the new context, return to persist state changes
```

Comment:
what if there is no timeoutNode/timeoutFlow ?  we don't throw anymore >

Reply:
Yes, it will not throw an error since that, in my opinion, can't be considered an error not to have a timeout flow/node; instead, if there is no jump, return the event to have its state saved (If we modify the state in the timeout hook for example) or if there was a jump in the timeout hook, process the event.